src/status_im2/contexts/shell/activity_center/events.cljs

```diff
@@ -156,14 +156,20 @@
                         :on-error   [:activity-center/process-notification-failure nil
                                      :notification/mark-all-as-read]}]})))
 
-(rf/defn mark-all-as-read-success
-  {:events [:activity-center.notifications/mark-all-as-read-success]}
-  [{:keys [db]}]
+(defn merge-chats
+  [db chats]
+  (reduce (fn [db' {:keys [chat-id] :as chat}] (update-in db' [:chats chat-id] merge chat)) db chats))
```

Comment:
@qfrank, I would usually add a few line breaks for improved readability. Lisps can get confusing with many expressions on the same line because we need to use indentation to make sense of the tree structure.

```clojure
(defn merge-chats
  [db chats]
  (reduce (fn [db' {:keys [chat-id] :as chat}]
            (update-in db' [:chats chat-id] merge chat))
          db
          chats))
```

But we can achieve the same more succinctly I think:

```clojure
(defn merge-chats-2
  [db chats]
  (update db :chats merge (utils.collection/index-by :chat-id chats)))
```

This gives the same results, which could be a unit test ;)

```clojure
(let [db    {:chats {"0x1" {:chat-id "0x1"}
                     "0x2" {:chat-id "0x2"}
                     "0x3" {:chat-id "0x3"}}}
      chats [{:chat-id "0x1" :clock 11}
             {:chat-id "0x3" :clock 33}]]
  (= (merge-chats db chats)
     (merge-chats-2 db chats)))
```